publisher-sdk/src/main/java/com/criteo/publisher/network/PubSdkApi.java

```diff
@@ -134,7 +136,15 @@ public InputStream executeRawGet(URL url) throws IOException {
   }
 
   public void postCsm(@NonNull MetricRequest request) throws IOException {
-    URL url = new URL(buildConfigWrapper.getCdbUrl() + "/csm");
+    postToCdb("/csm", request);
+  }
+
+  public void postLogs(@NonNull List<RemoteLogRecords> request) throws IOException {
+    postToCdb("/inapp/logs", request);
+  }
+
+  private void postToCdb(@NonNull String apiPath, @NonNull Object request) throws IOException {
```

Comment:
Do you think it is possible to refactor `loadCdb` to use `postToCdb` as well? (feels a bit weird to have a `postToCdb` method not used by another method that also makes a POST request to CDB).

Reply:
Unfortunately there is few differences between the bid request and the csm/logs request:
- bid request is logged (ok we can give a boolean to control this)
- bid request implies a response, moreover the response cannot just be read via the `JsonSerializer`, it is still using the old `JSONObject` with manual deserialization
- csm/logs does not read the response (just the HTTP status code)

Because of such differences, I don't think we can really use `postToCdb` from `loadCdb`

Reply:
If we want to factorize the `PubSdkApi`, then `loadConfig` and `loadCdb` are good matches. But again the issue is: logging (not a big issue) and the manual deserialization of `CdbResponse` (bigger issue).
If the deserialization issue is solved, then we could imagine a method like `postToCdbAndRead`.

Reply:
Ok indeed.